Sources/TuistKit/Utils/ManifestGraphLoader.swift

```diff
@@ -99,32 +99,44 @@ public final class ManifestGraphLoader: ManifestGraphLoading {
         // Load Plugins
         let plugins = try await loadPlugins(at: path)
 
+        // Load Workspace
+        var allManifests = try recursiveManifestLoader.loadWorkspace(at: path)
+        let isSPMProjectOnly = allManifests.projects.isEmpty
```

Comment:
The connection between `allManifests.projects.isEmpty` and `isSPMProjectOnly` is a strong assumption that we can break easily. What does it mean `isSPMProjectOnly`?

Reply:
I meant if no projects are found at this stage, the only remaining possibility is to have an SPM only project.
https://github.com/tuist/tuist/pull/6011

Reply:
I am open to suggestions for better login / naming.
However I noticed Marek did some work in the same area in another PR https://github.com/tuist/tuist/pull/6060 and this PR might not be valid anymore.

Reply:
> I meant if no projects are found at this stage, the only remaining possibility is to have an SPM only project.

In that case, since `loadWorkspace` returns a `Graph` instance, I think it'd make sense to make that more explicit:

```swift
var type: GraphType {
  if allManifests.projects.isEmpty {
    .swiftPackage
  } else {
    .tuistProject
  } 
}
```

That way, graph consumers can use that explicit API instead of leaking the graph's internals through `allManifests.projects.isEmpty`. I'm looping in @fortmarek who has worked on that piece recently and can provide a better advise here.

Reply:
Actually `recursiveManifestLoader.loadWorkspace(at: path) ` returns a `TuistLoader.LoadedWorkspace` which is an intermediary data representation of the workspace before trying to load and merge the `Package.swift` manifest, and converting finally to a `Graph`. It is unfortunate that following sequence of code

```swift
let manifest = try swiftPackageManagerGraphLoader.load(
                packagePath: packagePath,
                packageSettings: loadedPackageSettings
            )
```

will flag the install error if install was not run. But this is not necessary if  intermediary `allManifest` has no use of external dependences. And it is a chicken and egg problem in the way the code is architected.

I agree, that my solution is a workaround for the moment, and a more in depth refactoring would be necessary, to have the final graph loaded and flag errors if needed. The refactoring and rearchitecting  I am not able to tackle for the moment with my knowledge of Tuist codebase, but I've seen in #6060 changes which could maybe integrate also this feature, although I would say it is a bug.